### PR TITLE
Re-enable OOUI confirm tags dialog

### PR DESCRIPTION
**Pull request by egardner**

### CODE REVIEW

**Comment by egardner** [comment]
This is working, but we'll need to figure out a way to write a meaningful test for the publish functionality now that the ConfirmTagsDialog acts as an intermediary between the ImageCard's "publish" button and the Vuex publish action.

Mocking the entire ConfirmTagsDialog class via the `__mocks__` folder is probably the best way to proceed. Looks like the way Jest wants us to do this is to add a `__mocks__` folder directly next to the file we'd be grabbing, so `resources/widgets/__mocks__/ConfirmTagsDialog.js`. I'll try to get this working on Monday.

**Comment by egardner** [comment]
Mocking the ConfirmTagsDialog widget wasn't necessary – it just needed to be exposed as `this.confirmTagsDialog` so that it could be manipulated programmatically during testing.

File: resources/components/ImageCard.vue [hunk 1]
```diff
@@ -176,7 +177,20 @@ module.exports = {
...
 		 */
 		onPublish: function () {
-			this.publishTags();
+			var windowManager = new OO.ui.WindowManager();
+
+			// We need to expose confirmTagsDialog on the vm so that we can access it in testing
+			this.confirmTagsDialog = new ConfirmTagsDialog( {
+				tagsList: this.confirmedSuggestions.map( function ( tag ) {
+					return tag.text;
+				} ).join( ' ,' ),
```
Comment (AnneTee) [inline resources/components/ImageCard.vue]: @egardner This should be `.join( ', ' )`

File: resources/components/ImageCard.vue [hunk 2]
```diff
@@ -176,7 +177,20 @@ module.exports = {
 		 * lives in Vuex, so all we need to do is trigger things here.
 		 */
 		onPublish: function () {
-			this.publishTags();
+			var windowManager = new OO.ui.WindowManager();
+
+			// We need to expose confirmTagsDialog on the vm so that we can access it in testing
```
Comment (AnneTee) [inline resources/components/ImageCard.vue:182]: Can you add a comment in the code explaining this about more?

File: tests/jest/ImageCard.test.js [hunk 3]
```diff
@@ -83,10 +83,10 @@ describe( 'ImageCard', () => {
 
 		const wrapper = VueTestUtils.mount( ImageCard, { store, localVue } );
 		const publishButton = wrapper.find( '.wbmad-action-buttons__publish' );
-
 		expect( actions.publishTags ).not.toHaveBeenCalled();
 
 		publishButton.trigger( 'click' );
+		wrapper.vm.confirmTagsDialog.emit( 'confirm' );
```
Comment (AnneTee) [inline tests/jest/ImageCard.test.js:89]: Can you add a comment explaining this too?

**Review by AnneTee — CHANGES_REQUESTED** [review]
@egardner Just one small change needed, and I thought it would be helpful to have some comments in the code explaining what's going on with the vm since it's new for most people, myself included, and vm stands for something else in the minds of most programmers!

File: resources/components/ImageCard.vue (hunk 2 above)
Comment (egardner) [reply resources/components/ImageCard.vue:182]: I have some additional comments in the follow-up PR that enables the "Add custom tag" dialog: https://github.com/egardner/mediawiki-extensions-MachineVision/pull/39/files#diff-492d669d28bf88c47e5aa363e279c44eR186-R234

That PR might be a better place to flesh out the rest of the comments more if needed.

File: tests/jest/ImageCard.test.js (hunk 3 above)
Comment (egardner) [reply tests/jest/ImageCard.test.js:89]: See above – I'll add more documentation in #39

**Review by AnneTee — APPROVED** [review]

**Pull request merged**